File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        t = sorted(nums1 + nums2)
        m = len(nums1)
        n = len(nums2)
        if m == 0 and n == 0:
            return 0
        if ((m+n) % 2 == 0):
            i = (m+n) // 2
            return float((t[i]+t[i-1])/2)
        else:
            i = (m+n)// 2
            return float(t[i])
        

File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int i = 0, j = 0;
        int sa[] = new int[nums1.length + nums2.length];
        int k = 0;
        while ( i < nums1.length && j < nums2.length) {
            if (nums1[i] < nums2[j]) {
                sa[k] = nums1[i];
                k++;
                i++;
            } else {
                sa[k] = nums2[j];
                k++;
                j++;
            }
        }
        
        while (i < nums1.length) {
            sa[k] = nums1[i];
            k++;
            i++;
        }
        
        while (j < nums2.length) {
            sa[k] = nums2[j];
            k++;
            j++;
        }
        
        int length = nums1.length + nums2.length;
        System.out.println(length);
        if (length % 2 == 0) {
            double a = sa[length/2];
            double b = sa[(length/2) - 1];
            System.out.println(a);
            System.out.println(b);
            return (a+b) / 2;
        } else {
            return sa[length/2];
        }
    }
}